python/triton/language/semantic.py

```diff
@@ -1367,9 +1367,14 @@ def _validate_dtype(dtype, allowed_types, operand_name):
     assert lhs_rank == rhs_rank == 2 or lhs_rank == rhs_rank == 3, f"Both inputs must be either 2D or 3D; (lhs: {lhs.shape} vs rhs: {rhs.shape})"
     assert lhs.shape[-1].value == rhs.shape[
         -2].value, f"First input shape ({lhs.shape}) and second input shape {rhs.shape} are not compatible for matmul (second index of first shape ({lhs.shape[-1].value}) must be equal to first index of second shape ({rhs.shape[-2].value})"
-    assert lhs.shape[-2].value >= 16 and lhs.shape[-1].value >= 16 \
-        and rhs.shape[-1].value >= 16, \
-        f"All non-batch values in both first input shape ({lhs.shape}) and second input shape ({rhs.shape}) must be >= 16!"
+    if "gfx9" in builder.options.arch:
```

Comment:
Yeah in theory any size should always work in the frontend, and in the backend we decide how to lower it down. However, it's worth noticing that if I try smaller layouts (e.g., 8x8x8) this gives back mismatches. I think the reason why we don't have mismatches for 16x16x8 is because we emit the mfma.

Reply:
So, one possible step for this PR is to investigate why our non-accelerated lowering fails. If we can sort this out, I think we can make any size to work and remove this limitation.

Reply:
As we agreed, in this PR we don't want to change the backend, so for now we only safely add support for 16x16x8. Anything smaller than that would require additional backend work

Reply:
Please note, I ended up checking for `arch` with `hasattribute()`. This is because to test for the type I had to `import` HIPOptions in `semantic.py` and this might be something we don't want